client/dashboard/profile-wizard/steps/industry.js

```diff
@@ -100,14 +117,28 @@ class Industry extends Component {
 				<Card>
 					<div className="woocommerce-profile-wizard__checkbox-group">
 						{ Object.keys( industries ).map( slug => {
+							const selectedIndustry = find( selected, { slug: slug } );
+
 							return (
-								<CheckboxControl
-									key={ slug }
-									label={ industries[ slug ] }
-									onChange={ () => this.onChange( slug ) }
-									checked={ selected.includes( slug ) }
-									className="woocommerce-profile-wizard__checkbox"
-								/>
+								<div key={ `div-${ slug }` }>
+									<CheckboxControl
+										key={ `checkbox-control-${ slug }` }
+										label={ industries[ slug ].label }
+										onChange={ () => this.onChange( slug ) }
+										checked={ selectedIndustry || false }
+										className="woocommerce-profile-wizard__checkbox"
+									/>
+									{ industries[ slug ].use_description &&
+										selectedIndustry && (
+											<TextControl
+												key={ `text-control-${ selectedIndustry.slug }` }
+												label={ industries[ selectedIndustry.slug ].description_label }
+												value={ selectedIndustry.detail || textInputListContent[ slug ] || '' }
```

Comment:
Ah, perhaps this is the culprit for the resetting field value?

Reply:
Yes! This bug happens when the data is already saved and the `Other` option is unselected and reselected. Now the old value is updating every time the input changes.